PhysicsTools/PatAlgos/plugins/PATTauProducer.cc

```diff
@@ -79,16 +80,36 @@ PATTauProducer::PATTauProducer(const edm::ParameterSet& iConfig)
   if (addTauID_) {
     // read the different tau ID names
     edm::ParameterSet idps = iConfig.getParameter<edm::ParameterSet>("tauIDSources");
-    std::vector<std::string> names = idps.getParameterNamesForType<edm::InputTag>();
+    std::vector<std::string> names = idps.getParameterNamesForType<edm::ParameterSet>();
+    std::map<std::string, IDContainerData> idContainerMap;
     for (std::vector<std::string>::const_iterator it = names.begin(), ed = names.end(); it != ed; ++it) {
-      tauIDSrcs_.push_back(NameTag(*it, idps.getParameter<edm::InputTag>(*it)));
+      edm::ParameterSet idp = idps.getParameter<edm::ParameterSet>(*it);
+      int wpidx = idp.getParameter<int>("workingPointIndex");
+      edm::InputTag tag = idp.getParameter<edm::InputTag>("inputTag");
+      if (wpidx == -99) {
+        tauIDSrcs_.push_back(NameTag(*it, tag));
+      } else {
+        std::map<std::string, IDContainerData>::iterator it2;
+        it2 = idContainerMap
+                  .insert(std::pair<std::string, IDContainerData>(tag.label() + tag.instance(),
+                                                                  IDContainerData(tag, std::vector<NameWPIdx>())))
+                  .first;
+        it2->second.second.push_back(NameWPIdx(*it, wpidx));
```

Comment:
But this would not save 'tag' in IDContainerData when initializing a map entry. This would require extra handling which is not really nicer imo. The declaration of the iterator could actually be avoided if both commands are chained. But I'm not sure whether this is preferable in terms of readability.

Reply:
ah, ok, I missed the non-default assignment in the insert; it sounds like it's two lines then.
some ```auto``` and list initializations  like ```{a,b}``` instead of ```pair<TypeA,TypeB>(a,b)``` can still be less verbose